ompi/mpi/java/c/Makefile.am

```diff
@@ -44,7 +46,7 @@ libmpi_java_la_SOURCES = \
 	mpi_Status.c \
 	mpi_Win.c
 
-libmpi_java_la_LIBADD = $(top_builddir)/ompi/libmpi.la
+libmpi_java_la_LIBADD = -ldl $(top_builddir)/ompi/libmpi.la
```

Comment:
Does this naked `-ldl` need to be protected?  E.g.:

* is `-ldl` available on every platform?  (I don't think it is...?)
* does configure elsewhere choose specifically not to link against `dl` when we `--disable-dlopen`?
* what happens if you `-ldl` in a static build?